### PR TITLE
Refactor `LayerSwitcher` and `MapComponent`

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This refactors the `LayerSwitcher` and the `MapComponent`. Both are transformed to react function components.

The `LayerSwitcher` is also restructured to have less internal functions and variables.
Two new optional properties where added ('identifierProperty', 'labelProperty') and the typing of the layers property is looser then before.

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the
  [BSD-2-Clause](https://github.com/terrestris/react-geo/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/react-geo/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [Code of Conduct](https://github.com/terrestris/react-geo/blob/main/CODE_OF_CONDUCT.md).
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

File: src/LayerSwitcher/LayerSwitcher.tsx [hunk 1]
```diff
@@ -47,96 +49,77 @@ export type LayerSwitcherProps = OwnProps & React.HTMLAttributes<HTMLDivElement>
  * @class LayerSwitcher
  * @extends React.Component
  */
-export class LayerSwitcher extends React.Component<LayerSwitcherProps, LayerSwitcherState> {
+// export class LayerSwitcher extends React.Component<LayerSwitcherProps, LayerSwitcherState> {
```
Comment (dnlkoch) [inline src/LayerSwitcher/LayerSwitcher.tsx:52]:
```suggestion
```

File: src/LayerSwitcher/LayerSwitcher.tsx [hunk 2]
```diff
@@ -26,15 +24,19 @@ export interface OwnProps {
   /**
    * The layers to be available in the switcher.
    */
-  layers: ArrayTwoOrMore<OlLayerTile<OlTileSource> | OlLayerGroup>;
+  layers: OlLayerBase[];
```
Comment (dnlkoch) [inline src/LayerSwitcher/LayerSwitcher.tsx:27]: Is a layer group supported as input layer? If not we might want to set the type to `ol/layer/Layer`.

File: src/Map/MapComponent/MapComponent.tsx [hunk 3]
```diff
@@ -1,102 +1,44 @@
...
-        id={mapDivId}
-        {...passThroughProps}
-      >
-        {children}
-      </div>;
+  const refCallback = useCallback((ref: HTMLDivElement) => {
+    if (!map) {
+      return;
+    }
+    if (ref == null) {
```
Comment (dnlkoch) [inline src/Map/MapComponent/MapComponent.tsx:22]:
```suggestion
    if (ref === null) {
```

File: src/LayerSwitcher/LayerSwitcher.tsx (hunk 2 above)
Comment (KaiVolland) [reply src/LayerSwitcher/LayerSwitcher.tsx:27]: LayerGroup is still supported.

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**